src/gcode-parser.ts

```diff
@@ -60,13 +60,69 @@ type MoveCommandParamName = 'x' | 'y' | 'z' | 'r' | 'e' | 'f' | 'i' | 'j';
 type MoveCommandParams = {
   [key in MoveCommandParamName]?: number;
 };
+
+export enum Code {
+  G0 = 'G0',
+  G1 = 'G1',
+  G2 = 'G2',
+  G3 = 'G3',
+  G20 = 'G20',
+  G21 = 'G21',
+  G28 = 'G28',
+  T0 = 'T0',
+  T1 = 'T1',
+  T2 = 'T2',
+  T3 = 'T3',
+  T4 = 'T4',
+  T5 = 'T5',
+  T6 = 'T6',
+  T7 = 'T7'
+}
 export class GCodeCommand {
+  public code?: Code;
   constructor(
     public src: string,
     public gcode: string,
     public params: CommandParams,
     public comment?: string
-  ) {}
+  ) {
+    this.code = this.match(gcode);
+  }
+
+  match(gcode: string): Code {
+    switch (gcode) {
+      case 'g0':
+      case 'g00':
+        return Code.G0;
+      case 'g1':
+      case 'g01':
+        return Code.G1;
+      case 'g2':
+      case 'g02':
+        return Code.G2;
+      case 'g3':
+      case 'g03':
+        return Code.G3;
+      case 't0':
+        return Code.T0;
+      case 't1':
+        return Code.T1;
+      case 't2':
+        return Code.T2;
+      case 't3':
+        return Code.T3;
+      case 't4':
+        return Code.T4;
+      case 't5':
+        return Code.T5;
+      case 't6':
+        return Code.T6;
+      case 't7':
+        return Code.T7;
+      default:
+        return undefined;
+    }
+  }
```

Comment:
There must be a better way. The idea here is to parse the text gcode command into a predictable format. I used an enum to make it easy to type, but there's something smelly about this. Any idea on how to change that?

Reply:
hmmm, instead of Enums, union types of string literals could be used too: 
`type Command = 'g0' | g2 | .. `

It's a little cleaner but suffers from the same problems. 

Another way is to us `as const`:
`const commands = ['g0', 'g1', ...] as const` 

This creates a type from the whole array literal. But since it already is a value we can use it to compare other values against, and use functions like toLowercase to allow for variants. I think it is the cleanest and DRYest way in terms of bothe types and run-time code.

Reply:
We could also interpret `t` as the command and the number after it as a param. I was just lazy I guess when I first wrote it ;-)

Reply:
You just gave me an idea. Any type of command is a letter + a number. This could help with parsing `g1`/`G01`, if we decompose it.